### PR TITLE
[llvm_instcount] Leaderboard Submission: GATv2 with DD-PPO

### PR DESCRIPTION
**Pull request by anthony0727**
This adds entrypoint for model learnt from Programl observations.

Programl encoded with graph neural network, and align tasks in as much parallelism as possible to remove temporal correlation(just like replay buffer from Deep-Q Network). The model is optimized with PPOv2 loss.

The datasets used to train are ['cbench-v1', 'mibench-v1', 'blas-v0', 'npb-v0'] with limited node/edge counts(due to our shortage of resouce).

Anticipating better result with additional experiments(this has been halted from our internal server maintanance, and will be resumed when done).

### CODE REVIEW

**Comment by ChrisCummins** [comment]
Hi @anthony0727, fantastic! Great to see a GNN-backed RL implementation. Couple of things:

- Your `results.csv` file is missing columns. The easiest way to make sure you populate all of the columns in the right order is to use the [`compiler_gym.CompilerEnvStateWriter`](https://compilergym.com/compiler_gym/compiler_gym.html#compiler_gym.CompilerEnvStateWriter) to create your CSV file.
- There is trailing whitespace in your README.md. You can either strip that using your editor, or use our [pre-commit](https://github.com/facebookresearch/CompilerGym/blob/development/.pre-commit-config.yaml#L2-L17) hooks to do this automatically.

Also, be sure to add an entry to the main README.md with your entry!

BTW, the vocab file that I think you are looking for can be found here: https://zenodo.org/record/4247595

Cheers,
Chris

**Comment by anthony0727** [comment]
Hey @ChrisCummins, resovled two bullets you told me!
Again, `ghostscript` was excluded in the `result.csv`, due to resource shortage

BTW,
Can the ranking in the main README.md be updated with new experiments? (though ranking is not about everything, I observed some better results, but it's gone with the server issue)

Thanks!
Anthony

File: README.md [hunk 1]
```diff
@@ -146,8 +146,9 @@ environment on the 23 benchmarks in the `cbench-v1` dataset.
...
+| Anthony. W. Jung                                 | GATv2 + DD-PPO | [write-up](leaderboard/llvm_instcount/gatv2_ddppo/README.md), [results](leaderboard/llvm_instcount/gatv2_ddppo/result.csv) | 2022-06 | 351.611s        | 1.047×                       |
 | Facebook                                         | Random search (t=60) | [write-up](leaderboard/llvm_instcount/random_search/README.md), [results](leaderboard/llvm_instcount/random_search/results_p125_t60.csv) | 2021-03 | 91.215s         | 1.045×                       |
-| Facebook                                         | e-Greedy search (e=0.1) | [write-up](leaderboard/llvm_instcount/e_greedy/README.md), [results](leaderboard/llvm_instcount/e_greedy/results_e10.csv) | 2021-03 | 152.579s        | 1.041×                       |
+| Facebook                                         | e-Greedy search (e=0.1) | [write-up](leaderboard/llvm_instcount/e_greedy/README.md), [results](leaderboard/llvm_instcount/e_greedy/results_e10.csv) | 2021-03 | 351.611s        | 1.041×                       |
```
Comment (ChrisCummins) [inline README.md:151]: Is this correct?

**Comment by ChrisCummins** [comment]
Hi Anthony,

Thanks for the fixes 🙂 

> BTW, Can the ranking in the main README.md be updated with new experiments? (though ranking is not about everything, I observed some better results, but it's gone with the server issue)

Yes, just send us a PR when you have the new results to update your position on the leaderboard.

One small comment about changing the walltime reported for greedy search. Other than, LGTM!

Cheers,
Chris

File: README.md (hunk 1 above)
Comment (anthony0727) [reply README.md:151]: Sorry, I calculated again, and it's `258.149s`

**Pull request merged**